aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -15,220 +15,445 @@ uid: fundamentals/servers/kestrel
+
+* Set **AllowInvalid** to `true` to permit the use of invalid certificates (for example, self-signed certificates).
+* **HttpsDefaultCert** indicates use of the cert defined under **Certificates** > **Default** or the development certificate.
+
+```json
+{
+"Kestrel": {
+  "EndPoints": {
+    "Http": {
+      "Url": "http://localhost:5000"
+    },
+
+    "HttpsInlineCertFile": {
+      "Url": "https://localhost:5001",
+      "Certificate": {
+        "Path": "<path to .pfx file>",
+        "Password": "<certificate password>"
+      }
+    },
+
+    "HttpsInlineCertStore": {
+      "Url": "https://localhost:5002",
+      "Certificate": {
+        "Subject": "<subject; required>",
+        "Store": "<certificate store; defaults to My>",
+        "Location": "<location; defaults to CurrentUser>",
+        "AllowInvalid": "<true or false; defaults to false>"
+      }
+    },
+
+    "HttpsDefaultCert": {
+      "Url": "https://localhost:5003"
+    },
+
+    "Https": {
+      "Url": "https://*:5004",
+      "Certificate": {
+      "Path": "<path to .pfx file>",
+      "Password": "<certificate password>"
+      }
+    }
+    },
+    "Certificates": {
+      "Default": {
+        "Path": "<path to .pfx file>",
+        "Password": "<certificate password>"
+      }
+    }
+  }
+}
+```
+
+An alternative to using **Path** and **Password** for the **Certificates** > **Default** certificate specification is to specify the certificate using the certificate store fields:
+
+```json
+"Default": {
+"Subject": "<subject; required>",
+"Store": "<cert store; defaults to My>",
+"Location": "<location; defaults to CurrentUser>",
+"AllowInvalid": "<true or false; defaults to false>"
+}
+```
+
+Schema notes:
+
+* Endpoints names are case-insensitive. For example, `HTTPS` and `Https` are valid.
+* The `Url` parameter is required for each endpoint. The format for this parameter is the same as the top-level `Urls` configuration parameter except that it's limited to a single value.
+* These endpoints replace those defined in the top-level `Urls` configuration rather than adding to them. Endpoints defined in code via `Listen` are cumulative with the endpoints defined in the configuration section.
+* The `Certificate` section is optional. If the `Certificate` section isn't specified, the defaults defined in earlier scenarios are used. If no defaults are available, the server throws an exception and fails to start.
+* The `Certificate` section supports both **Path**&ndash;**Password** and **Subject**&ndash;**Store** certificates.
+* Any number of endpoints may be defined in this way so long as they don't cause port conflicts.
+* `serverOptions.Configure(context.Configuration.GetSection("Kestrel"))` returns a `KestrelConfigurationLoader` with an `.Endpoint(string name, options => { })` method that can be used to supplement a configured endpoint's settings:
+
+  ```csharp
+  serverOptions.Configure(context.Configuration.GetSection("Kestrel"))
```

Comment:
You can also directly access KestrelServerOptions.ConfigurationLoader to keep iterating on the existing loader, such as the one provided by WebHost.CreatedDeafaultBuilder.